ProcessOptimizer/learning/gaussian_process/gpr.py

```diff
@@ -327,33 +325,65 @@ def predict(self, X, return_std=False, return_cov=False,
 
         else:  # Predict based on GP posterior
             K_trans = self.kernel_(X, self.X_train_)
-            y_mean = K_trans.dot(self.alpha_)    # Line 4 (y_mean = f_star)
+            y_mean = K_trans @ self.alpha_    # Line 4 (y_mean = f_star)
             # undo normalisation
             y_mean = self.y_train_std_ * y_mean + self.y_train_mean_
```

Comment:
Over in scikit-learn ([here](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/gaussian_process/_gpr.py)), this line reads:
`y_mean = self._y_train_std * y_mean + self._y_train_mean`
Note the lack of underscores at the end. If I remember correctly, the values stored in self._y_train_mean represent the values computed from the original data, while the value self._y_train_mean_ (with the underscore) represent the value in the model after fitting? The correct way to undo the normalization is to transform back using the original mean and std, so I just want to make sure that this is what we actually do.

Reply:
Or is this maybe just the consequence of calling 'super(GaussianProcessRegressor, self).fit(X, y)'?

Reply:
Yes, It is because we leave most parts of the math to sklearns own implementation of the GPR. But as long as we support old sklearn-versions, we have this code to ensure that normalization of y-space is done correctly.

Reply:
Right, so just to be 100 % sure that I understand it: The present use of `std_` and `mean_` is actually fully equivalent to the scikit-lean implementation, but in our case `self` is just wrapped an extra time around the object?